### PR TITLE
Mm delete region

### PR DESCRIPTION
**Pull request by MMatusu**
Delete region button function and disable events setup.

### CODE REVIEW

**Comment by MMatusu** [comment]
Update of the region panel is necessary, alongside with name structures update -  by now, delete works fine only for regions of the same dimension.

**Comment by MMatusu** [comment]
cfg.diagram.regions.regions is in methods treated as a dictionary, yet it is a list, except for inserting ones -> TODO: Adjust insert, add, etc. methods, change data structure.
Note: This is not an issue as long as the list is not anyhow reduced and all indices are consequent.

File: src/LayerEditor/ui/panels/diagram.py [hunk 1]
```diff
@@ -106,38 +106,60 @@ def __init__(self, parent=None):
...
+            dim = 1
+        elif isinstance(obj, Polygon):
+            dim = 2
+        if dim is not None:
+            reg_id = cfg.main_window.regions.get_current_region()
+            if not reg_id == 0:
+                region = cfg.diagram.regions.regions[reg_id]
+                pas = region.cmp_shape_dim(layer_id, dim)
+                if pas:
+                    self.regionsUpdateRequired.emit()
```
Comment (jbrezmorf) [inline src/LayerEditor/ui/panels/diagram.py]: Explain why we need update for this case.

File: src/LayerEditor/ui/panels/diagram.py [hunk 2]
```diff
@@ -106,38 +106,60 @@ def __init__(self, parent=None):
...
+        self.addItem(obj)
+        dim = None
+        layer_id = cfg.diagram.regions.current_layer_id
+        if isinstance(obj, Point):
+            dim = 0
+        elif isinstance(obj, Line):
+            dim = 1
+        elif isinstance(obj, Polygon):
+            dim = 2
+        if dim is not None:
```
Comment (jbrezmorf) [inline src/LayerEditor/ui/panels/diagram.py]: Add comments.
Seems that this should rather be an assert, and possibly we need check for `obj is None` in beginning of function.

File: src/LayerEditor/ui/panels/diagram.py (hunk 1 above)
Comment (MMatusu) [reply src/LayerEditor/ui/panels/diagram.py]: The whole function can be reduced in the way of complementary function remove_graphical_object(). The whole segment only filters unnecessary updates of the region panel.
The main purpose of the update signal on object addition is disabling the remove button once the region is used.

**Comment by jbrezmorf** [comment]
Bug: Face and Edge regions can be deleted even if in use.

File: src/LayerEditor/ui/mainwindow.py [hunk 3]
```diff
@@ -268,9 +268,11 @@ def set_topology(self):
         self.regions.set_topology(cfg.diagram.topology_idx)
         
     def _update_regions(self):
-        """Update region for set shape"""
-        regions =self.diagramScene.selection.get_selected_regions(cfg.diagram)
-        self.regions.select_current_regions(regions)
+        """Update region panel, eventually set tab according to the selection in diagram"""
+        regions = self.diagramScene.selection.get_selected_regions(cfg.diagram)
+        if regions[0]:
```
Comment (jbrezmorf) [inline src/LayerEditor/ui/mainwindow.py]: - 'select_current_regions' should be named 'select_current_region' and just single region should be passed
- For selection of single region, this should be selected in region panel. For more selected regions it was proposed to display None region, but regio of the last selected-deselected object can be used.

Comment (MMatusu) [reply src/LayerEditor/ui/mainwindow.py]: That is an original function out of this pull request scope. However, a new branch is created for that pull request purpose.

**Pull request merged**